tests/providers/test_shell.py

```diff
@@ -80,5 +86,6 @@ def test_valid_env(self):
         assert result["env"] == {"FOO": "BAR"}
 
     def test_invalid_env(self):
-        with pytest.raises(ApiError):
+        with pytest.raises(ApiError) as e:
             parse_provider_config("shell", {"command": "/usr/bin/true", "env": "FOO"})
+        assert e.value.message == 'Option "env" is not a valid type for provider: shell'
```

Comment:
hmmm what's setting `message`? usually when I see `.message` I assume python 2.x -- typically you'd either use `str(e.value)` or `e.args[0]`

Reply:
```
class ApiError(Exception):
    def __init__(self, message, name=None, status_code=400):
        self.message = message
```

Reply:
GOnna refactor this, I agree the .message throws me off with python 2.x.

Reply:
hah it's also not pickleable -- but maybe that's fine :P

Reply:
Oh, lol, it's already available in __str__.